homeassistant/components/nextbus/sensor.py

```diff
@@ -0,0 +1,250 @@
+"""NextBus sensor."""
+import logging
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import CONF_NAME
+from homeassistant.const import DEVICE_CLASS_TIMESTAMP
+from homeassistant.helpers.entity import Entity
+from homeassistant.util.dt import utc_from_timestamp
+
+REQUIREMENTS = ['py_nextbus==0.1.2']
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_AGENCY = 'agency'
+CONF_ROUTE = 'route'
+CONF_STOP = 'stop'
+
+ICON = 'mdi:bus'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_AGENCY): cv.string,
+    vol.Required(CONF_ROUTE): cv.string,
+    vol.Required(CONF_STOP): cv.string,
+    vol.Optional(CONF_NAME): cv.string,
+})
+
+
+def get_predictions(stops, agency, client=None):
+    """Return predictions for provided stops."""
+    if client is None:
+        from py_nextbus import NextBusClient
+        client = NextBusClient(output_format='json')
+
+    return client.get_predictions_for_multi_stops(stops, agency)
+
+
+def validate_value(value_name, value, value_list):
+    """Validate tag value is in the list of items and logs error if not."""
+    valid_values = {
+        v['tag']: v['title']
+        for v in value_list
+    }
+    if value not in valid_values:
+        _LOGGER.error(
+            'Invalid %s tag `%s`. Please use one of the following: %s',
+            value_name,
+            value,
+            ', '.join(
+                '{}: {}'.format(title, tag)
+                for tag, title in valid_values.items()
+            )
+        )
+        return False
+
+    return True
+
+
+def validate_tags(agency, route, stop, client=None):
+    """Validate provided tags."""
+    if client is None:
```

Comment:
Remove this case. It's only used in testing. We can create a client from the mocked library and pass that in the test.